### PR TITLE
Update FormContext::errors() to handle nested errors.

### PR DESCRIPTION
**Pull request by beporter**
Adjust error collection to properly handle nested validators. Partially addresses #7532.

### CODE REVIEW

**Comment by beporter** [comment]
Give me a few minutes to check/supplement the tests.

File: src/View/Form/FormContext.php [hunk 1]
```diff
@@ -135,8 +135,9 @@ public function hasError($field)
     public function error($field)
     {
         $errors = $this->_form->errors();
-        if (isset($errors[$field])) {
-            return array_values($errors[$field]);
+        $errors = \Cake\Utility\Hash::extract($errors, $field);
```
Comment (dereuromark) [inline src/View/Form/FormContext.php]: Shouldnt we not use FQNS here?

Comment (beporter) [inline src/View/Form/FormContext.php]: Sure, I just started with the patch I posted in #7532. I need a little time to clean things up a bit.

**Comment by beporter** [comment]
This still doesn't address `isRequired()` like @markstory mentioned, nor does it test an `addNestedMany()` validator at all, but it's a start.

**Comment by lorenzo** [comment]
Thanks!

**Comment by markstory** [comment]
@beporter Fixing FormContext might be enough. I'm not sure many people use nested validators with the ORM.

**Pull request merged**